From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Wed, 16 Jun 2021 20:34:01 +0200
Subject: [PATCH] dt-bindings: phy: meson8b-usb2: Add support for reading the
 ID signal

The first USB PHY is OTG capable which means that the USB "ID" signal is
routed to the PHY (instead of an GPIO input). Add support for the
gpio-controller and #gpio-cells properties so the value of the "ID"
signal can be read as a GPIO (from the PHY).

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 .../bindings/phy/amlogic,meson8b-usb2-phy.yaml        | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/Documentation/devicetree/bindings/phy/amlogic,meson8b-usb2-phy.yaml b/Documentation/devicetree/bindings/phy/amlogic,meson8b-usb2-phy.yaml
index 03c4809dbe..c354919c01 100644
--- a/Documentation/devicetree/bindings/phy/amlogic,meson8b-usb2-phy.yaml
+++ b/Documentation/devicetree/bindings/phy/amlogic,meson8b-usb2-phy.yaml
@@ -6,6 +6,12 @@ $schema: "http://devicetree.org/meta-schemas/core.yaml#"
 
 title: Amlogic Meson8, Meson8b, Meson8m2 and GXBB USB2 PHY
 
+description: |
+  Amlogic Meson8, Meson8b, Meson8m2 and GXBB embed two USB2 PHYs.
+  Typically the USB "ID" signal (used for dual role operation) is
+  routed to the first PHY. In that case the value can be read out
+  as a GPIO.
+
 maintainers:
   - Martin Blumenstingl <martin.blumenstingl@googlemail.com>
 
@@ -31,6 +37,11 @@ properties:
       - const: usb_general
       - const: usb
 
+  '#gpio-cells':
+    const: 2
+
+  gpio-controller: true
+
   resets:
     minItems: 1
 
